flightsql/src/main/java/io/deephaven/server/flightsql/FlightSqlResolver.java

```diff
@@ -0,0 +1,1916 @@
+            case COMMAND_GET_CROSS_REFERENCE_TYPE_URL:
+                return GET_CROSS_REFERENCE_HANDLER;
+            case COMMAND_STATEMENT_SUBSTRAIT_PLAN_TYPE_URL:
+                return STATEMENT_SUBSTRAIT_PLAN_HANDLER;
+            case COMMAND_PREPARED_STATEMENT_UPDATE_TYPE_URL:
+                return PREPARED_STATEMENT_UPDATE_HANDLER;
+            case COMMAND_GET_XDBC_TYPE_INFO_TYPE_URL:
+                return GET_XDBC_TYPE_INFO_HANDLER;
+        }
+        throw error(Code.UNIMPLEMENTED, String.format("command '%s' is unknown", typeUrl));
+    }
+
+    private TicketHandler ticketHandlerForResolve(SessionState session, Any message) {
+        final String typeUrl = message.getTypeUrl();
+        if (TICKET_STATEMENT_QUERY_TYPE_URL.equals(typeUrl)) {
+            final TicketStatementQuery ticketStatementQuery = unpackOrThrow(message, TicketStatementQuery.class);
+            final TicketHandler ticketHandler = queries.get(ticketStatementQuery.getStatementHandle());
+            if (ticketHandler == null) {
+                throw error(Code.NOT_FOUND,
+                        "Unable to find FlightSQL query. FlightSQL tickets should be resolved promptly and resolved at most once.");
+            }
+            if (!ticketHandler.isOwner(session)) {
+                // We should not be concerned about returning "NOT_FOUND" here; the handleId is sufficiently random that
+                // it is much more likely an authentication setup issue.
+                throw permissionDeniedWithHelpfulMessage();
+            }
+            return ticketHandler;
+        }
+        final CommandHandler commandHandler = commandHandler(session, typeUrl, false);
+        try {
+            return commandHandler.initialize(message);
+        } catch (StatusRuntimeException e) {
+            // This should not happen with well-behaved clients; or it means there is a bug in our command/ticket logic
+            throw error(Code.INVALID_ARGUMENT,
+                    "Invalid ticket; please ensure client is using an opaque ticket", e);
+        }
+    }
+
+    private Table executeSqlQuery(SessionState session, String sql) {
+        // See SQLTODO(catalog-reader-implementation)
+        // final QueryScope queryScope = sessionState.getExecutionContext().getQueryScope();
+        final QueryScope queryScope = ExecutionContext.getContext().getQueryScope();
+        // noinspection unchecked,rawtypes
+        final Map<String, Table> queryScopeTables =
+                (Map<String, Table>) (Map) queryScope.toMap(queryScope::unwrapObject, (n, t) -> t instanceof Table);
+        final TableSpec tableSpec = Sql.parseSql(sql, queryScopeTables, TicketTable::fromQueryScopeField, null);
+        // Note: this is doing io.deephaven.server.session.TicketResolver.Authorization.transform, but not
+        // io.deephaven.auth.ServiceAuthWiring
+        try (final SafeCloseable ignored = LivenessScopeStack.open(new LivenessScope(), true)) {
```

Comment:
Same result, more common call pattern for it:
```suggestion
        try (final SafeCloseable ignored = LivenessScopeStack.open()) {
```

Reply:
If you take the SingletonLivenessManager approach, you would put the LSS.open() call in `executeImpl` and then you would manage it with the singleton manager before letting it close:

```java
            try (final SafeCloseable ignored = LivenessScopeStack.open()) {
                table = executeSqlQuery(session, sql);
                manager.manage(table);
            } catch (SqlParseException e) {
```